---
title: "Repet_1"
author: "Parth"
date: "2024-04-23"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Install all packages 
```{r}
install.packages("tidyverse")
library(tidyverse)
library(ggplot2)
library(dplyr)
library(DESeq2)

```
```{r}
library(AnnotationDbi)
library(org.Hs.eg.db)
```
Obtaining the data and mearging the files for Dseq2 analysis 

```{r}
library(tximport)
library(DESeq2)
library(readr)

# List the directories in the salmon_quant directory
dirs <- list.files("C:/Users/parth/OneDrive/Desktop/rnaseq_data/salmon_quant")

# List the files in the salmon_quant directory
quant_files <- list.files("C:/Users/parth/OneDrive/Desktop/rnaseq_data/salmon_quant",
                          pattern = "quant.sf",
                          recursive = TRUE,
                          full.names = TRUE)

# Assign names to quant_files based on directory names
names(quant_files) <- dirs

# Read transcript-to-gene mapping CSV file
tx2gene <- read_csv("C:/Users/parth/OneDrive/Desktop/rnaseq_data/tx2gene.csv", col_names = FALSE)

# Import transcript-level abundance estimates using tximport
txi <- tximport(files = quant_files,
                type = "salmon",
                tx2gene = tx2gene,
                ignoreTxVersion = TRUE)

# Read sample information from TSV file
sampleinfo <- read_tsv("C:/Users/parth/OneDrive/Desktop/rnaseq_data/meta_data/sampleInfo_corrected.txt")

# Create DESeqDataSet object
dds <- DESeqDataSetFromTximport(txi, 
                                colData = sampleinfo,
                                design = ~Treated)

```


Now view all the files 
```{r}
library(tidyverse)
View(quant_files)
View(dds)
tibble(tx2gene)
View(sampleinfo)

```
So for now we have imported all the files and we are good to go, to perform Dseq 2 analysis we have to convert the condtion variable to make R treat them as numberical values First check
```{r}
type(dds$condition)
```
Now convert this in numnber--- did not understand why this did not work?
```{r}
dds$condition <- as.factor(dds$condition)
type(dds$condition)
```
create a directory and save everything in R objects file 
```{r}
dir.create("Robjects",showWarnings = FALSE)
saveRDS(dds,file = "Robjects/dds.rds")
```

Workflow 
```{r}
colData(dds)
```
Now desgin the experiment (try conditon here) and perform differential seq analysis 
```{r}
design(dds)<-~condition
de_conditon<-DESeq(dds)
results(de_conditon)

```
```{r}
results(de_conditon,tidy = TRUE)
```
Sorting data properly for analysis 
```{r}
library(dplyr)
library(tidyverse)
library(DESeq2)
results(de_conditon,tidy = TRUE)%>% arrange(padj)%>% head(n=100)
```

Now the main workflow, create directory, files and perform DSEq2 major analysis 
```{r}
dir.create("de_analysis",showWarnings = FALSE)
results(de_conditon,tidy = TRUE)%>%arrange(desc(padj>0.05))%>%write.csv("conditional_analysis.csv")
```

Now save an RDS copy of this file important for graph making 
```{r}
saveRDS(de_conditon,file = "conditional_analysis")

```
check the file 
```{r}
results(de_conditon)
library(tidyverse)
de_conditon
```
Plot an MA plot 
```{r}
plotMA(de_conditon)
```
Check individual datapoints 
```{r}
plotCounts(dds,"ENSG00000158258",intgroup = "condition")
```
Changing direction of contrast 
```{r}
# Assuming you're working with the limma package for differential expression analysis

# Specify the contrasts you're interested in
contrasts_of_interest <- c("condition", "CTR", "TGF")

# Assuming `de_condition` represents your design matrix or contrast matrix
# Call `results()` function with the necessary arguments
results <- results(de_conditon, contrast = contrasts_of_interest)

# Now you can proceed with further operations on `results`, such as summarizing or visualizing

```
After obtaining the results of de_condition, designing the matrix and obtaining the required p value these values are now annotated with  Biocmanager package 
```{r}
install.packages("BiocManager")
install.packages("org.Hs.eg.db")
library(BiocManager)
library(org.Hs.eg.db)



```
```{r}
install.packages("org.Hs.eg.db")
library(org.Hs.eg.db)
columns(org.Hs.eg.db)
```
Now we add anootation filters using keytypes and key filters to check the data
```{r}
keytypes(org.Hs.eg.db)
```

```{r}
keys(org.Hs.eg.db)
```

We can;t undertstand the numerical output and we have to assign them ENSEMBL (D (here)) 
```{r}
keys(org.Hs.eg.db,keytype = "ENSEMBL")[1:5]
```
```{r}
keys(org.Hs.eg.db,keytype = "ENTREZID")[1:5]
```
Both ENSEMBL and ENTREZID are assigned a gene ID according to the respective databases and a unique code is provided to them 
Now we assign gene names and its symbol in the ENSEMBL database 
```{r}
select(org.Hs.eg.db,keys = "ENSG00000158258",
       keytype = "ENSEMBL",columns=c("SYMBOL","GENENAME"))
```
Now annotate this database 
```{r}
AnnotationDbi::select(org.Hs.eg.db,keys ="ENSG00000158258",keytype = "ENSEMBL",column=c("SYMBOL","GENENAME"))
```
Now annotate the whole dataset 
```{r}
library(DESeq2)
DESeq(dds)
```

```{r}
anno<-AnnotationDbi::select(org.Hs.eg.db,keys = rownames(dds),columns = c("SYMBOL","GENENAME"),keytype = "ENSEMBL")
tibble(anno)

```

```{r}
results_annotated<-results(de_conditon,tidy=TRUE)%>%left_join(anno,by=c("row"="ENSEMBL"))
head(results_annotated)
```
Write the csv file 
```{r}
write.csv(results_annotated,file = "De_condition_analysis.csv",row.names = FALSE)
saveRDS(results_annotated,file = "Robjects/de_condition.rds")
```
Exporting normailesed counts 
```{r}
dds <- estimateSizeFactors(dds) 
countmatrix<-counts(dds,normalized=TRUE)
head(countmatrix)

```
```{r}
#creating a ggplot to check the de_conditon database and see if we can generate ggplots accordingly this has been generated with dds and de_condition modified database
ggplot(data = results_annotated) +
  geom_point(mapping = aes(x = GENENAME, y = stat,colour=stat))
```
Creating volcano plots and DSeq 2 analysis 
```{r}
library(ggplot2)
library(DESeq2)
results_condition<-readRDS("C:/Users/parth/OneDrive/Desktop/repet_1/Repeat_1/Robjects/de_condition.rds")
results_condition %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(padj))) + geom_point()
```
```{r}
results_condition %>% 
  mutate(Significant = padj < 0.05 & abs(log2FoldChange) > 2) %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(padj), col=Significant)) + geom_point()
```

Applying the ifelse function 
```{r}
N <- 10
if(!require(ggrepel))install.packages("ggrepel")
top_genes <- dplyr::slice(results_condition, 1:N) %>% pull(row)

results_condition %>% 
  mutate(Label = ifelse(row %in% top_genes, SYMBOL, "")) %>%  
  ggplot(aes(x = log2FoldChange, y = -log10(padj), label=Label)) + geom_point(alpha=0.4) + geom_text(col="blue")

```

```{r}
head(results_condition)
```

Generate heatmaps of the code given 
```{r}
library(pheatmap)
top_genes<-dplyr::slice(results_condition,1:10)%>%pull(row)
vsd<-vst(dds)
pheatmap(assay(vsd)[top_genes,])
```
Now it is ridiculous that you have to rememeber the ENSEMBL ID names so you can annotate them by using the annotationdbi:select function
before that you merge the sample info with dds before you do that convert it to as dataframe 
```{r}
sampleInfo <- as.data.frame(colData(dds)[,c("condition","Treated")])#generating data from sampleinfo and merfing with dds data
#now generating pheatmap 
sampleInfo <- as.data.frame(colData(dds)[,c("condition","Treated")])

pheatmap(assay(vsd)[top_genes,],
         annotation_col = sampleInfo,scale = "row")
```
```{r}
#now for the finale 
gene_labels<-dplyr::slice(results_condition,1:N)%>%pull(SYMBOL)
#Now label the dataset for heat map
pheatmap(assay(vsd)[top_genes,],
         annotation_col = sampleInfo,
         labels_row = gene_labels,
         scale="row")
#SO hereby according to the data using condition to design the matrix we found that FCR expression is highest in non treated samples in CTR expressing dataset
#remember the design function is used to focus your dataset on  one point and design your experiment around it 
```

Now pathway analysis 
Threshold based gene testing to generate molecular pathways, here we check for the ECM pathways in cancer using the org.hs.eb.db database  
```{r}
library(org.Hs.eg.db)
pathway_genes <- AnnotationDbi::select(org.Hs.eg.db,
                                       keys = "GO:0030198",
                                       keytype = "GO",
                                       columns="ENSEMBL") %>% pull(ENSEMBL)


```

```{r}
go_table <- mutate(results_condition, 
                   inPathway = row %in% pathway_genes,
                   isDE = padj < 0.05 & abs(log2FoldChange) > 1)
go_table
```
pull out and check which values are true or false for your dataset and perform statistical test
```{r}
table(go_table$inPathway,go_table$isDE)
chisq.test(table(go_table$inPathway, go_table$isDE))
```
ORA analysis with gene cluster profiler 
```{r}
library(clusterProfiler)
universe <- results_condition %>% pull(row)
sigGenes <- results_condition %>% 
  filter(padj < 0.05, !is.na(row)) %>% pull(row)

enrich_go <- enrichGO(
  gene= sigGenes,
  OrgDb = org.Hs.eg.db,
  keyType = "ENSEMBL",
  ont = "BP",
  universe = universe,
  qvalueCutoff = 0.05,
  readable=TRUE
)
enrich_go%>%data.frame()
View(enrich_go)
```
```{r}
dotplot(enrich_go,showCategory=10)
```
Apply for emaplot 
```{r}
enrich_go <- enrichplot::pairwise_termsim(enrich_go)
emapplot(enrich_go)
```
```{r}
enrichplot::upsetplot(enrich_go)

```

GESA analysis: Gene set enrichment set analysis 
```{r}
ranked_genes <- results_condition %>% 
  arrange(desc(stat)) %>% 
  filter(!is.na(stat))
  
geneList <- pull(ranked_genes, stat)
names(geneList) <- pull(ranked_genes, row)
  
gse_GO  <- gseGO(geneList = geneList,
        OrgDb = org.Hs.eg.db,
        ont = "BP",keyType = "ENSEMBL")
```
performing plot analysis 
```{r}
ridgeplot(gse_GO)
```
```{r}
enrichplot::upsetplot(gse_GO)
```

```{r}
gseaplot(gse_GO,geneSetID = "GO:0002286")

```
Interactive visualisation using genetonic 
```{r}
library(GeneTonic)
design(dds) <- ~condition
de <- DESeq(dds)
##Don't use the tidy=TRUE option so the output stays as a DESeq object
res_de <- results(de,contrast = c("condition", "TGF","CTR"))
```
Using Shake function: converting everything in common format 
```{r}
res_enrich <- shake_enrichResult(enrich_go)
```


```{r}
anno_df <- AnnotationDbi::select(org.Hs.eg.db,keys=rownames(dds),columns="SYMBOL",keytype = "ENSEMBL")  %>% 
  dplyr::rename(gene_id = ENSEMBL,gene_name=SYMBOL)
```
```{r}
GeneTonic(dds,
  res_de,
  res_enrich,
  anno_df)#create a new website for genetonic website 
```

